include/swift/Parse/SyntaxTransformer.h

```diff
@@ -0,0 +1,68 @@
+//===--- SyntaxTransformer.h ----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_PARSE_SYNTAXTRANSFORMER_H
+#define SWIFT_PARSE_SYNTAXTRANSFORMER_H
+
+#include "swift/AST/ASTContext.h"
+#include "swift/AST/Expr.h"
+#include "swift/Syntax/SyntaxNodes.h"
+#include "llvm/ADT/DenseMap.h"
+
+namespace swift {
+/// Transforms Syntax nodes into AST nodes.
+class SyntaxTransformer {
```

Comment:
`SyntaxTransformer` is too generic for a name, it seems as if it is a utility class to accommodate any kind of transformation, but in reality it has a very specific purpose. I like @brentdax's suggestion for `ASTGen` instead.

Reply:
Done.